pkg/sql/distsql_plan_bulk.go

```diff
@@ -37,11 +40,12 @@ func (dsp *DistSQLPlanner) SetupAllNodesPlanningWithOracle(
 	evalCtx *extendedEvalContext,
 	execCfg *ExecutorConfig,
 	oracle replicaoracle.Oracle,
+	localityFilter roachpb.Locality,
 ) (*PlanningCtx, []base.SQLInstanceID, error) {
-	if dsp.codec.ForSystemTenant() {
-		return dsp.setupAllNodesPlanningSystem(ctx, evalCtx, execCfg, oracle)
+	if dsp.codec.ForSystemTenant() && !localityFilter.NonEmpty() {
+		return dsp.setupAllNodesPlanningSystem(ctx, evalCtx, execCfg, oracle, localityFilter)
```

Comment:
Why does having a locality filter force us into tenant planning?

Reply:
_non-deprecated_ planning, because initially I only added locality-filtering/selection to the new (tenant-compatible) planning implementations and not the gossip ones.

But I ended up plumbing the filtering though both the standard (tenant) and deprecated (system) all nodes setup (if not the actual span planning), so I can remove this one I guess. The other has to stay though.